src/__swaps__/screens/Swap/components/ReviewPanel.tsx

```diff
@@ -157,13 +158,15 @@ export function ReviewPanel() {
     });
   }, [navigate]);
 
-  const openGasExplainer = useCallback(() => {
+  const openGasExplainer = useCallback(async () => {
+    const nativeAsset = await getNativeAssetForNetwork(
```

Comment:
i think the `getNativeAssetForNetwork` util might be overkill here. looks like we only need the native asset's icon url, symbol, network name, and colors, which you can get from the synchronous util `getNetworkNativeAsset` in the same file

Reply:
agreed but that function relies on user assets. What happens when a user doesn't have the native asset for the network?

Reply:
ok true